forwarder/update_handlers/ca_update_handler.py

```diff
@@ -41,16 +37,15 @@ def __init__(
         self._logger = get_logger()
         self._producer = producer
         self._output_topic = output_topic
-        self._cached_update: Optional[Tuple[ReadNotifyResponse, int]] = None
-        self._output_type: Any = None
+        self._cached_update: Optional[ReadNotifyResponse] = None
         self._repeating_timer = None
         self._cache_lock = Lock()
 
         try:
-            self._message_publisher = schema_publishers[schema]
+            self._message_serialiser = schema_serialisers[schema](pv_name)
         except KeyError:
             raise ValueError(
-                f"{schema} is not a recognised supported schema, use one of {list(schema_publishers.keys())}"
+                f"{schema} is not a recognised supported schema, use one of {list(schema_serialisers.keys())}"
```

Comment:
Instead of `except`, before you call this, you may check if the key is there. If not raise a value error. Raising within except should be avoided.

Reply:
True, done.